Add blog for Retroactive Default StorageClass move to GA (#41982)
kubernetes/website: content/en/blog/_posts/2023-08-15-retroactive-default-storage-class-ga.md

---
layout: blog
title: "Kubernetes v1.28: Retroactive Default StorageClass move to GA"
date: 2023-08-15
slug: retroactive-default-storage-class-ga
---

**Author:** Roman Bednář (Red Hat)

Announcing graduation to General Availability (GA) - Retroactive Default StorageClass Assignment in Kubernetes v1.28!

> We are thrilled to announce that the "Retroactive Default StorageClass Assignment" feature,

Reviewer: Who are “we” here? Kubernetes SIG Storage perhaps?

Author: I meant Kubernetes community but it's not too apparent for the reader. Since there are links to SIG Storage below we can use Kubernetes SIG storage here instead.

introduced as an alpha in Kubernetes v1.25, has now graduated to GA and is officially part of the Kubernetes v1.28 release.

> This enhancement brings a significant improvement to how default StorageClasses are assigned to PersistentVolumeClaims (PVCs).

Reviewer: ```suggestion
This enhancement brings a significant improvement to how default
[StorageClasses](/docs/concepts/storage/storage-classes/) are assigned to PersistentVolumeClaims (PVCs).
```

With this feature enabled, you no longer need to create a default StorageClass first and then a PVC to assign the class.
Instead, any PVCs without a StorageClass assigned will now be retroactively updated to include the default StorageClass.
This enhancement ensures that PVCs no longer get stuck in an unbound state, and storage provisioning works seamlessly,
even when a default StorageClass is not defined at the time of PVC creation.

## What changed?

The PersistentVolume (PV) controller has been modified to automatically assign a default StorageClass to any unbound

> PersistentVolumeClaim with the storageClassName set to null. Additionally, the PersistentVolumeClaim admission within

Reviewer: ```suggestion
PersistentVolumeClaim with the `storageClassName` not set. Additionally, the PersistentVolumeClaim
admission validation mechanism within
```
In the API response, I think the field is omitted when not set. Have I got that right?

Author: Correct, and the previous post which we reference here describes the difference between "not set" and `""` in more detail. Saying "not set" here is a more accurate summary of the new behavior for sure.

the API server has been adjusted to allow changing values from an unset state to an actual StorageClass name.

## How to use it?

As this feature has graduated to GA, there's no need to enable a feature gate anymore.
Simply make sure you are running Kubernetes v1.28 or later, and the feature will be available for use.

> For more details on using the Retroactive Default StorageClass Assignment, refer to the Kubernetes documentation [here](/docs/concepts/storage/persistent-volumes/#retroactive-default-storageclass-assignment).

Reviewer: ```suggestion
For more details, read about
[default StorageClass assignment](/docs/concepts/storage/persistent-volumes/#retroactive-default-storageclass-assignment) in the Kubernetes documentation.
```

---
layout: blog
title: "Kubernetes v1.28: Retroactive Default StorageClass move to GA"

> date: 2023-08-15

Reviewer: /hold

I want to get it confirmed that this is the data that release comms have assigned. OK to unhold if it is.

Reviewer 2: Hi, release lead shadow for 1.28 here. 

According to the [tracking board](https://github.com/orgs/kubernetes/projects/140/views/6) the blog has been scheduled for August 18th 2023. Please update the date accordingly, thanks!

> date: 2023-08-15

Reviewer 3: Hi @RomanBednar , here Comms shadow, could you please edit for the scheduled date?
```suggestion
date: 2023-08-18
```

Author: @krol3 Done and squashed.
